Dynatrace/dynatrace-operator
# Remove DockerConfig struct

## Description

As we no longer need the `DockerConfig` and no longer is it actually used (because it was replaced by the keychain + new library), it should be removed.

## How can this be tested?

Check that the `VersionStatus` of the dynakube is properly set
Check that the `codeModulesImage` field can be used and functions.

## Checklist
